web/src/service-worker/index.ts

```diff
@@ -66,6 +103,40 @@ async function immichAsset(url: URL) {
   }
 }
 
+async function appResources(url: URL, event: FetchEvent) {
```

Comment:
A major part of this seems to overlap with `ssr`. I feel like you can extract a bunch of this out into a private function or call `ssr` inside `appResources`